log.go

```diff
@@ -47,12 +45,21 @@ func Logger(system string) *ZapEventLogger {
 	return &ZapEventLogger{system: system, SugaredLogger: *logger}
 }
 
-// ZapEventLogger implements the EventLogger and wraps a go-logging Logger
+// ZapEventLogger implements the EventLogger and wraps a zap Sugared Logger.
 type ZapEventLogger struct {
 	zap.SugaredLogger
 	system string
 }
 
+// SetFieldsOnLogger adds the provided key value args as fields to
+// the embedded zap.SugaredLogger. These fields are separate to those
+// that are provided to the logger via SetFieldsOnAllLoggers and these
+// only last for the life time of this particular ZapEventLogger instance.
+// Note: the fields will be passed to any children loggers of this logger.
+func (zel *ZapEventLogger) SetFieldsOnLogger(name string, args ...interface{}) {
+	zel.SugaredLogger = *zel.With(args...)
+}
```

Comment:
How is this supposed to be used? This isn't going to be thread-safe.

I'd expect users to just call `zel.With(...)` and use the new logger.

Reply:
> I'd expect users to just call zel.With(...) and use the new logger.

The issue with this is that `zel.With(...)` returns a `zap.SugaredLogger` and not a `log.ZapEventLogger` which is the logger that users will be initializing and using throughout their apps. Calling `zel.With(...)` essentially un-embeds the zap logger...

Reply:
We can always define our own `With` that returns a `ZapEventLogger`, right?

(also, this _still_ isn't thread-safe and there's no way to make it thread-safe without performance issues)